apps/antalmanac/src/components/Calendar/Toolbar/CustomEventDialog/LocationSelector.tsx

```diff
@@ -0,0 +1,89 @@
+import Checkbox from '@material-ui/core/Checkbox';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import FormGroup from '@material-ui/core/FormGroup';
+import React, { PureComponent } from 'react';
+import { Paper, Tab, Tabs, TextField } from '@material-ui/core';
+import { styled, Theme, withStyles } from '@material-ui/core/styles';
+import { ClassNameMap, Styles } from '@material-ui/core/styles/withStyles';
+import { Autocomplete } from '@material-ui/lab';
+
+import Building from '../../../RightPane/Map/static/building';
+import buildingCatalogue from '../../../RightPane/Map/static/buildingCatalogue';
+import locations from '../../../RightPane/SectionTable/static/locations.json';
+
+const styles: Styles<Theme, object> = {
+    tabContainer: {
+        zIndex: 1000,
+        marginLeft: '15%',
+        marginRight: '15%',
+        marginTop: '8px',
+        marginBottom: '8px',
+        position: 'relative',
+    },
+    searchBarContainer: {
+        minWidth: '60%',
+        position: 'relative',
+        marginLeft: '15%',
+        marginRight: '15%',
+        zIndex: 1000,
+        alignItems: 'center',
+    },
+};
+
+const StyledTabs = styled(Tabs)({
+    minHeight: 0,
+});
+
+const StyledTab = styled(Tab)({
+    minHeight: 'auto',
+    minWidth: '10%',
+    padding: 0,
+});
+
+
+interface LocationSelectorProps {
+    handleSearch: (event: React.ChangeEvent<unknown>, value: Building | null) => void;
+    previousOption: string;
+}
+
+class LocationSelector extends PureComponent<LocationSelectorProps>  {
```

Comment:
At first glance, this component seems really similar in functionality to the `RightPane/Map/MapMenu` component. Can we combine their functionality into a single component that can be reused in both the Map and Custom Event creator menus? One potential solution would be replacing `LocationSelector` with a `MapMenu` that has a custom `handleSearch` function passed in. Or if these are different enough that they need to be separate, let me know why and include it as a comment in the source code.

Reply:
I changed this class to a location selector class, so wherever we want to add a location selector, we can use this class. This is now used by the MapMenu & CustomEventDialog. I delete the unnecessary part of this class.